flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveDialectITCase.java

```diff
@@ -807,6 +807,29 @@ public void testShowPartitions() throws Exception {
         assertTrue(partitions.toString().contains("dt=2020-04-30 01:02:03/country=china"));
     }
 
+    @Test
+    public void testStatementVariableSubstitution() {
+        // test system variable for substitution
+        System.setProperty("k1", "v1");
+        List<Row> result =
+                CollectionUtil.iteratorToList(
+                        tableEnv.executeSql("select '${system:k1}'").collect());
+        assertEquals("[+I[v1]]", result.toString());
+
+        // test env variable for substitution
+        String classPath = System.getenv("CLASSPATH");
+        result =
+                CollectionUtil.iteratorToList(
+                        tableEnv.executeSql("select '${env:CLASSPATH}'").collect());
+        assertEquals(String.format("[+I[%s]]", classPath), result.toString());
+
+        // test hive conf variable for substitution
+        result =
+                CollectionUtil.iteratorToList(
+                        tableEnv.executeSql("select '${hiveconf:common-key}'").collect());
```

Comment:
How do users dynamically set hiveconf variables in the flink table api? `common-key` is a variable that exists in the test hive-site.xml

Reply:
Currently, we have no way to dynamically set hiveconf variables in the flink table api.  AFAK, hive only supports set hiveconf variables in cli or init a hive session with hiveconf variables.